### ISSUE TITLE
Missing typescript declaration files for @mapbox/point-geometry , @mapbox/vector-tile

### ISSUE THREAD

**Issue by saosebastiao** [maplibre/maplibre-gl-js#1096]
**maplibre-gl-js version**: 2.1.7

**browser**: N/A

### Steps to Trigger Behavior

run `tsc` on project using maplibre-gl dependency

my tsconfig.json:
```
{
  "compilerOptions": {
    "target": "ES6",
    "useDefineForClassFields": true,
    "lib": ["DOM", "DOM.Iterable", "ESNext"],
    "allowJs": false,
    "skipLibCheck": false,
    "esModuleInterop": false,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "module": "ESNext",
    "moduleResolution": "Node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "jsx": "react-jsx"
  },
  "include": ["./src"]
}
```

### Expected Behavior

tsc runs without errors

### Actual Behavior

```
node_modules/maplibre-gl/dist/maplibre-gl.d.ts:3:19 - error TS7016: Could not find a declaration file for module '@mapbox/point-geometry'. '/Users/DToone2/ws/auction_visualizer/node_modules/@mapbox/point-geometry/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/mapbox__point-geometry` if it exists or add a new declaration (.d.ts) file containing `declare module '@mapbox/point-geometry';`

3 import Point from '@mapbox/point-geometry';
                    ~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/maplibre-gl/dist/maplibre-gl.d.ts:5:52 - error TS7016: Could not find a declaration file for module '@mapbox/vector-tile'. '/Users/DToone2/ws/auction_visualizer/node_modules/@mapbox/vector-tile/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/mapbox__vector-tile` if it exists or add a new declaration (.d.ts) file containing `declare module '@mapbox/vector-tile';`

5 import { VectorTileFeature, VectorTileLayer } from '@mapbox/vector-tile';
                                                     ~~~~~~~~~~~~~~~~~~~~~


Found 2 errors.
```

**Comment by HarelM** [comment]
I don't think they are missing. They are configured in the package dependencies. It's probably depends on the build system you are using.

**Comment by kylebez** [comment]
This is also an issue for me. What is the solution?

Also can't find the namespace GeoJSON:

```
Error: node_modules/maplibre-gl/dist/maplibre-gl.d.ts:2650:13 - error TS2503: Cannot find namespace 'GeoJSON'.
2650  _geometry: GeoJSON.Geometry;
```

**Comment by wipfli** [comment]
Thanks for reporting.

Steps to reproduce this bug:

```
git clone git@github.com:wipfli/maplibre-typescript.git
cd maplibre-typescript
git checkout v2.1.7
nvm use 16
npm ci
npm run build
```

<details>
<summary>Generates this output:</summary>

```
➜  maplibre-typescript git:(797f3f9) npm run build         

> build
> tsc

node_modules/maplibre-gl/dist/maplibre-gl.d.ts:2650:13 - error TS2503: Cannot find namespace 'GeoJSON'.

2650  _geometry: GeoJSON.Geometry;
                 ~~~~~~~

node_modules/maplibre-gl/dist/maplibre-gl.d.ts:2655:18 - error TS2503: Cannot find namespace 'GeoJSON'.

2655  get geometry(): GeoJSON.Geometry;
                      ~~~~~~~

node_modules/maplibre-gl/dist/maplibre-gl.d.ts:2656:18 - error TS2503: Cannot find namespace 'GeoJSON'.

2656  set geometry(g: GeoJSON.Geometry);
                      ~~~~~~~

node_modules/maplibre-gl/dist/maplibre-gl.d.ts:4709:13 - error TS2503: Cannot find namespace 'GeoJSON'.

4709  features?: GeoJSON.Feature[];
                 ~~~~~~~

node_modules/maplibre-gl/dist/maplibre-gl.d.ts:4712:13 - error TS2503: Cannot find namespace 'GeoJSON'.

4712  features?: GeoJSON.Feature[];
                 ~~~~~~~

node_modules/maplibre-gl/dist/maplibre-gl.d.ts:5997:9 - error TS2503: Cannot find namespace 'GeoJSON'.

5997  _data: GeoJSON.GeoJSON | string;
             ~~~~~~~

node_modules/maplibre-gl/dist/maplibre-gl.d.ts:6017:16 - error TS2503: Cannot find namespace 'GeoJSON'.

6017  setData(data: GeoJSON.GeoJSON | string): this;
                    ~~~~~~~

node_modules/maplibre-gl/dist/maplibre-gl.d.ts:6033:65 - error TS2503: Cannot find namespace 'GeoJSON'.

6033  getClusterChildren(clusterId: number, callback: Callback<Array<GeoJSON.Feature>>): this;
                                                                     ~~~~~~~

node_modules/maplibre-gl/dist/maplibre-gl.d.ts:6059:94 - error TS2503: Cannot find namespace 'GeoJSON'.

6059  getClusterLeaves(clusterId: number, limit: number, offset: number, callback: Callback<Array<GeoJSON.Feature>>): this;
                                                                                                  ~~~~~~~

node_modules/maplibre-gl/dist/maplibre-gl.d.ts:6959:41 - error TS2503: Cannot find namespace 'GeoJSON'.

6959  setGeoJSONSourceData(id: string, data: GeoJSON.GeoJSON | string): void;
                                             ~~~~~~~


Found 10 errors.
```
</details>

**Comment by wipfli** [comment]
v2.0.5 passes without errors.

**Comment by wipfli** [comment]
`v2.1.0` produces these 6 errors:

```
➜  maplibre-typescript git:(main) ✗ npm run build                       

> build
> tsc

node_modules/maplibre-gl/dist/maplibre-gl.d.ts:2622:13 - error TS2503: Cannot find namespace 'GeoJSON'.

2622  _geometry: GeoJSON.Geometry;
                 ~~~~~~~

node_modules/maplibre-gl/dist/maplibre-gl.d.ts:2627:18 - error TS2503: Cannot find namespace 'GeoJSON'.

2627  get geometry(): GeoJSON.Geometry;
                      ~~~~~~~

node_modules/maplibre-gl/dist/maplibre-gl.d.ts:2628:18 - error TS2503: Cannot find namespace 'GeoJSON'.

2628  set geometry(g: GeoJSON.Geometry);
                      ~~~~~~~

node_modules/maplibre-gl/dist/maplibre-gl.d.ts:4588:13 - error TS2503: Cannot find namespace 'GeoJSON'.

4588  features?: GeoJSON.Feature[];
                 ~~~~~~~

node_modules/maplibre-gl/dist/maplibre-gl.d.ts:4591:13 - error TS2503: Cannot find namespace 'GeoJSON'.

4591  features?: GeoJSON.Feature[];
                 ~~~~~~~

node_modules/maplibre-gl/dist/maplibre-gl.d.ts:6511:41 - error TS2503: Cannot find namespace 'GeoJSON'.

6511  setGeoJSONSourceData(id: string, data: GeoJSON.GeoJSON | string): void;
                                             ~~~~~~~


Found 6 errors.

```

**Comment by wipfli** [comment]
v2.1.1 has the same 6 errors as v2.1.0

**Comment by wipfli** [comment]
v2.1.6 goes to 10 errors:

<details>
<summary>v2.1.6 errors</summary>
```
➜  maplibre-typescript git:(main) ✗ npm run build                       

> build
> tsc

node_modules/maplibre-gl/dist/maplibre-gl.d.ts:2650:13 - error TS2503: Cannot find namespace 'GeoJSON'.

2650  _geometry: GeoJSON.Geometry;
                 ~~~~~~~

node_modules/maplibre-gl/dist/maplibre-gl.d.ts:2655:18 - error TS2503: Cannot find namespace 'GeoJSON'.

2655  get geometry(): GeoJSON.Geometry;
                      ~~~~~~~

node_modules/maplibre-gl/dist/maplibre-gl.d.ts:2656:18 - error TS2503: Cannot find namespace 'GeoJSON'.

2656  set geometry(g: GeoJSON.Geometry);
                      ~~~~~~~

node_modules/maplibre-gl/dist/maplibre-gl.d.ts:4709:13 - error TS2503: Cannot find namespace 'GeoJSON'.

4709  features?: GeoJSON.Feature[];
                 ~~~~~~~

node_modules/maplibre-gl/dist/maplibre-gl.d.ts:4712:13 - error TS2503: Cannot find namespace 'GeoJSON'.

4712  features?: GeoJSON.Feature[];
                 ~~~~~~~

node_modules/maplibre-gl/dist/maplibre-gl.d.ts:5997:9 - error TS2503: Cannot find namespace 'GeoJSON'.

5997  _data: GeoJSON.GeoJSON | string;
             ~~~~~~~

node_modules/maplibre-gl/dist/maplibre-gl.d.ts:6017:16 - error TS2503: Cannot find namespace 'GeoJSON'.

6017  setData(data: GeoJSON.GeoJSON | string): this;
                    ~~~~~~~

node_modules/maplibre-gl/dist/maplibre-gl.d.ts:6033:65 - error TS2503: Cannot find namespace 'GeoJSON'.

6033  getClusterChildren(clusterId: number, callback: Callback<Array<GeoJSON.Feature>>): this;
                                                                     ~~~~~~~

node_modules/maplibre-gl/dist/maplibre-gl.d.ts:6059:94 - error TS2503: Cannot find namespace 'GeoJSON'.

6059  getClusterLeaves(clusterId: number, limit: number, offset: number, callback: Callback<Array<GeoJSON.Feature>>): this;
                                                                                                  ~~~~~~~

node_modules/maplibre-gl/dist/maplibre-gl.d.ts:6959:41 - error TS2503: Cannot find namespace 'GeoJSON'.

6959  setGeoJSONSourceData(id: string, data: GeoJSON.GeoJSON | string): void;
                                             ~~~~~~~


Found 10 errors.
```

**Comment by wipfli** [comment]
So this problem seems to have been introduced with `v2.1.0`.

**Comment by wipfli** [comment]
https://github.com/maplibre/maplibre-gl-js/compare/v2.0.5...v2.1.0 are the changes between the working and the breaking version.

**Comment by wipfli** [comment]
It looks like `@types/geojson` moved from a dependency to a devDependency in https://github.com/maplibre/maplibre-gl-js/commit/173d012f507bfbff4f66c032f5ecdd652dacda05

**Comment by wipfli** [comment]
@kylebez a workaround until we have fixed MapLibre GL JS is to run `npm install @types/geojson` in your npm project.

**Comment by wipfli** [comment]
Can confirm that this is fixed by `v2.1.8-pre.3`.
